Dockerfile.rocm.ubi

```diff
@@ -144,11 +150,12 @@ ENV VLLM_INSTALL_PUNICA_KERNELS=1
 RUN --mount=type=cache,target=/root/.cache/ccache \
     --mount=type=cache,target=/root/.cache/pip \
     --mount=type=cache,target=/root/.cache/uv \
+    ln -s /opt/vllm/lib64/python3.12/site-packages/torch/lib/libroctx64.so /opt/rocm/lib/libroctx64.so && \
```

Comment:
This fixes a cmake issue due to conflicting deps (cmake lookin in /opt/rocm/lib but libroctx64.so being shadowed by `/opt/vllm/...`